crates/router/src/core/payment_methods/cards.rs

```diff
@@ -4744,3 +5336,689 @@ pub async fn list_countries_currencies_for_connector_payment_method_util(
             .collect(),
     }
 }
+
+async fn get_pre_existing_pm(
+    state: &routes::SessionState,
+    req: &api::PaymentMethodCreate,
+    merchant_account: &domain::MerchantAccount,
+    key_store: &domain::MerchantKeyStore,
+    customer_id: &id_type::CustomerId,
+    pm_resp: &mut api::PaymentMethodResponse,
+    payment_method: diesel_models::PaymentMethod,
+) -> errors::RouterResult<diesel_models::PaymentMethod> {
+    let db = &*state.store;
+    let pm_update = storage::PaymentMethodUpdate::StatusUpdate {
+        status: Some(enums::PaymentMethodStatus::Inactive),
+    };
+
+    db.update_payment_method(payment_method, pm_update, merchant_account.storage_scheme)
```

Comment:
Already we are updating the payment_method then why we do have get_or_insert_payment_method below this?

Reply:
There are two different payment_methods involved here. For the `/save` endpoint, i.e. vault a payment_method using SDK, if the data for payment_method is duplicated, we deactivate the current record for the payment_method and fetch the already existing one.